iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -237,16 +244,20 @@ inline bool vector<T, Capacity>::resize(const uint64_t count, const Targs&... ar
     {
         while (count != m_size)
         {
-            emplace_back(args...);
+            IOX_DISCARD_RESULT(emplace_back(args...));
         }
     }
+    else
```

Comment:
I think we can rewrite it so that we do not need to use this empty `else`.

```cxx
    if (count < m_size)
    {
        clearFrom(count);
    }
    else
    {
        while(count != m_size)
        {
            IOX_DISCARD_RESULT(emplace_back(args...));
        }
    }

```